### ISSUE TITLE
Django out-of-the-box support

### ISSUE THREAD

**Issue by JakeCooper** [railwayapp/nixpacks#153]
Currently Django doesn't work out of the box because of the following error

```
Preparing metadata (setup.py): started
 
Preparing metadata (setup.py): finished with status 'error'
 
  error: subprocess-exited-with-error
 
× python setup.py egg_info did not run successfully.
│ exit code: 1
╰─> [23 lines of output]
running egg_info
creating /tmp/pip-pip-egg-info-vudn72c8/psycopg2.egg-info
writing /tmp/pip-pip-egg-info-vudn72c8/psycopg2.egg-info/PKG-INFO
writing dependency_links to /tmp/pip-pip-egg-info-vudn72c8/psycopg2.egg-info/dependency_links.txt
writing top-level names to /tmp/pip-pip-egg-info-vudn72c8/psycopg2.egg-info/top_level.txt
writing manifest file '/tmp/pip-pip-egg-info-vudn72c8/psycopg2.egg-info/SOURCES.txt'
 
Error: pg_config executable not found.
 
pg_config is required to build psycopg2 from source.  Please add the directory
containing pg_config to the $PATH or specify the full executable path with the
option:
 
python setup.py build_ext --pg-config /path/to/pg_config build ...
 
or with the pg_config option in 'setup.cfg'.
 
If you prefer to avoid building psycopg2 from source, please install the PyPI
'psycopg2-binary' package instead.
 
For further information please check the 'doc/src/install.rst' file (also at
<[https://www.psycopg.org/docs/install.html>](https://www.psycopg.org/docs/install.html%3E);).
 
[end of output]
 
note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed
 
× Encountered error while generating package metadata.
╰─> See above for output.
 
note: This is an issue with the package mentioned above, not pip.
hint: See above for details.
WARNING: You are using pip version 22.0.4; however, version 22.1.1 is available.
You should consider upgrading via the '/opt/venv/bin/python -m pip install --upgrade pip' command.
The command '/bin/bash -o pipefail -c python -m venv /opt/venv && . /opt/venv/bin/activate && pip install -r requirements.txt' returned a non-zero code: 1
 
Error: Docker build failed
```

I THINK this can be resolved by adding the nixpkg `postgresql`, but unsure

So, we need to check that and then add a provider for Django which, if it detects some Django-y things (manage.py in root? something else?) it'll grab that dep

**Comment by wyzlle** [comment]
Please share a minimal reproduction example.

**Comment by JakeCooper** [comment]
Our ideal with nixpacks is that users shouldn't have to change a lick of code. So, having someone need to change their deps isn't ideal

**Comment by coffee-cup** [comment]
Closed by #162
